Tooling/ScriptGenerator/ExtensionMethods.cs

```diff
@@ -0,0 +1,178 @@
+﻿// <copyright file="ExtensionMethods.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace ScriptGenerator;
+
+/// <summary>
+/// Provides extensions to improve code.
+/// </summary>
+public static class ExtensionMethods
+{
+    /// <summary>
+    /// Returns a value indicating whether or not the given <c>string</c> <paramref name="value"/>
+    /// only contains the given characters.
+    /// </summary>
+    /// <param name="value">The <c>string</c> to check.</param>
+    /// <param name="characters">The list of characters that are allowed in the given <paramref name="value"/>.</param>
+    /// <returns><c>true</c> if the given <paramref name="value"/> only contains the given <paramref name="characters"/>.</returns>
+    public static bool OnlyContains(this string value, IEnumerable<char> characters)
+        => !string.IsNullOrEmpty(value) && value.All(characters.Contains);
+
+    /// <summary>
+    /// Get service of type <typeparamref name="T"/> from the <see cref="IServiceProvider"/> and
+    /// handles an error when no service of type <typeparamref name="T"/> exists.
+    /// </summary>
+    /// <typeparam name="T">The type of service object to get.</typeparam>
+    /// <param name="provider">The <see cref="IServiceProvider"/> to retrieve the service object from.</param>
+    /// <returns>A service object of type <typeparamref name="T"/>.</returns>
+    /// <exception cref="System.InvalidOperationException">There is no service of type <typeparamref name="T"/>.</exception>
+    [ExcludeFromCodeCoverage]
+    public static T GetRequiredServiceAndHandleError<T>(this IServiceProvider provider)
+        where T : notnull
+    {
+        try
+        {
+            return provider.GetRequiredService<T>();
+        }
+        catch (InvalidOperationException e) when (e.Message.StartsWith("Unable to resolve service for type"))
+        {
+            var currentClr = Console.ForegroundColor;
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($"ERROR > {e.Message}");
+            Console.ForegroundColor = currentClr;
+            Environment.Exit(e.HResult);
+        }
+
+        return default!;
+    }
+
+    /// <summary>
+    /// Returns a value indicating whether or not the first occurence of the given <paramref name="beforeChar"/>
+    /// is located positionally before the given <paramref name="afterChar"/>.
+    /// </summary>
+    /// <param name="value">This <c>string</c> to search for the <paramref name="beforeChar"/> and <paramref name="afterChar"/>.</param>
+    /// <param name="beforeChar">The <c>char</c> that might be located before the <paramref name="afterChar"/>.</param>
+    /// <param name="afterChar">The <c>char</c> that might be located after the <paramref name="beforeChar"/>.</param>
```

Comment:
I don't like "might".   You can also use just "may be located...", or "can be located..."
```suggestion
    /// <param name="beforeChar">The <c>char</c> that may or may not be located before the <paramref name="afterChar"/>.</param>
    /// <param name="afterChar">The <c>char</c> that may or may not be located after the <paramref name="beforeChar"/>.</param>
```